src/config.rs

```diff
@@ -1,33 +1,140 @@
-use std::path::{Path, PathBuf};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    fs::File,
+    path::{Path, PathBuf},
+    process,
+};
+
+/// Restart conditions for a service.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "lowercase")]
+enum Restart {
+    Never,
+    Always,
+    Unexpected,
+}
 
-#[derive(Debug)]
 /// Configuration structure used to run a task.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+#[serde(default)]
 pub struct Config {
     /// Command to run.
     cmd: String,
     /// Number of processors this task should run with.
     // u16 is fine because we are not expecting a machine to run more
     // than 2^16 proc at any single time.
     numprocs: u16,
+    /// Program mask running under
+    umask: i32,
     /// Working directory of the task.
     workingdir: PathBuf,
+    /// Service starting at load time.
+    autostart: bool,
+    /// Service restarting when finished.
+    autorestart: Restart,
+    /// Autorestart callback trigger value.
+    exitcodes: Vec<i32>,
+    /// Restart attempts.
+    startretries: i32,
+    /// Delay before start.
+    starttime: i32,
+    /// Signal received by supervisord trigger service termination.
+    stopsignal: String,
+    /// Running period.
+    stoptime: i32,
+    /// Stdout logfile path.
+    stdout: PathBuf,
+    /// Stderr logfile path.
+    stderr: PathBuf,
+    /// Environment variable service is running into.
+    env: HashMap<String, String>,
```

Comment:
Do we need the environment to be copied here or could we just use [`std::env`](https://doc.rust-lang.org/std/env/index.html) ?

Reply:
The idea is to handle the environment for the process without having to use `env -i`. We cannot start each process this way because the config file loads them for us.
I would suggest we have:
* a "copy" mode with the current environment - an equivalent to run process like `TOTO=42 ls` - instead `taskmasterd` will do so for us.
* a "clean" mode equivalent to run `env -i TOTO=42 ls`.
From the `HashMap<String, String>` we capture TOTO=42 from config file and we will either start process with the whole existing std::env vars - "copy" mode - or not - "clean" mode.
What do you think of it ?

Reply:
We'll probably be running commands with `[std::process::Command](https://doc.rust-lang.org/std/process/struct.Command.html)`. Have a look at [this](https://doc.rust-lang.org/std/process/struct.Command.html#method.env_remove). I think we don't need to store the HashMap. What do you think ? :D